roboteam_ai/src/conditions/BallKickedToOurGoal.cpp

```diff
@@ -7,16 +7,17 @@ namespace rtt {
 namespace ai {
 
 BallKickedToOurGoal::BallKickedToOurGoal(std::string name, bt::Blackboard::Ptr blackboard)
-        :Condition(name, blackboard) { };
+        :Condition(std::move(name), std::move(blackboard)) { };
+
 bt::Node::Status BallKickedToOurGoal::update() {
-    roboteam_msgs::WorldBall ball = World::getBall();
+    auto ball = World::getBall();
     Vector2 goalCentre = Field::get_our_goal_center();
     double goalWidth = Field::get_field().goal_width;
     double margin = constants::BALL_TO_GOAL_MARGIN;
     Vector2 lowerPost = goalCentre + Vector2(0.0, - (goalWidth + margin));
     Vector2 upperPost = goalCentre + Vector2(0.0, goalWidth + margin);
-    Vector2 ballPos = ball.pos;
-    Vector2 ballPredPos = Vector2(ball.pos) + Vector2(ball.vel)*constants::BALL_TO_GOAL_TIME;
+    Vector2 ballPos = ball->pos;
+    Vector2 ballPredPos = Vector2(ballPos) + Vector2(ballPos)*constants::BALL_TO_GOAL_TIME;
```

Comment:
You changed a velocity to position here, please keep using the velocity (ball->vel)